### DISCUSSION TITLE
Idea: Allow second hub!?

### DISCUSSION THREAD

**Discussion by ctec76** [sfstar/hass-victron#132] [Ideas]
Hi,

First of all, many thanks to the main author and the contributors for doing this fantastic addon for HA via HACS, which saves a lot of time! 
Congrats on a job well done!

One thing I noticed so far is that it is only possible to add one hub, i.e., one Victron GX is possible as a main hub in the integration.
I am an installer with multiple GX installations at **one** location, i.e., multiple GX necessary in one HA.
There are several reasons/benefits of this approach which does not matter for the topic at hand.

With the current implementation of the hass-victron integration in HA, I am adding the main inverter, and as a workaround, I add the sensors to the other GX of the location by using manual modbus sensor registrations in the HA configuration.

It works, of course, but it would be great if the hass-victron integration supported more than one hub in the future!

Thanks!

All the very best!

**Comment by pos-ei-don** [comment]
+1

Do you have a complete modbus sensor list?
Can you share the yaml Code?

**Reply by ctec76** [reply]
Hi,

The entire sensor list is available on the Victron support side (as you probably know already).
https://www.victronenergy.com/support-and-downloads/technical-information

For my purposes, the following list is sufficient:
https://www.victronenergy.com/upload/documents/CCGX-Modbus-TCP-register-list-3.10.xlsx

To be able to include the **second** GX in HA via entities, I used the attached yaml code which works fine so far.
[config_part_modbus_Victron.txt](https://github.com/sfstar/hass-victron/files/12909821/config_part_modbus_Victron.txt)

The main GX is handled by the hass-Victron integration in HA via HACS, as stated in my original post.
Please note that for most installations, I only need basic sensor data in HA to be able to do the desired workflows.
See Victron Xls file for a full list.

Cheers!

**Comment by OTSkwebster** [comment]
Was there any progress on allowing a second hub to be added natively? I would be happy to fund the development.
